ui/vty/src/Ariadne/UI/Vty/Widget/Wallet.hs

```diff
@@ -152,26 +184,78 @@ handleWalletWidgetEvent = \case
           UiBalanceCommandFailure err -> BalanceResultError err
       other -> other
   UiCommandResult commandId (UiSendCommandResult result) -> do
-    walletSendResultL %= \case
+    use walletSendResultL >>= \case
       SendResultWaiting commandId' | commandId == commandId' ->
         case result of
-          UiSendCommandSuccess tr -> SendResultSuccess tr
-          UiSendCommandFailure err -> SendResultError err
-      other -> other
+          UiSendCommandSuccess tr -> do
+            walletSendResultL .= SendResultSuccess tr
+            walletSendPassL .= ""
+            walletSendOutputsL .= Map.empty
+            addOutput
+          UiSendCommandFailure err -> do
+            walletSendResultL .= SendResultError err
+      _ ->
+        return ()
   _ ->
     return ()
 
 ----------------------------------------------------------------------------
 -- Actions
 ----------------------------------------------------------------------------
 
+updateFocusList :: Monad m => StateT WalletWidgetState (StateT (WidgetInfo WalletWidgetState p) m) ()
+updateFocusList = do
+    outputs <- uses walletSendOutputsL Map.keys
+    lift $ setWidgetFocusList $
+      concat (outputFocuses <$> outputs) ++
+      [ WidgetNameWalletSendAdd
+      , WidgetNameWalletSendPass
+      , WidgetNameWalletSendButton
+      ]
+  where
+    outputFocuses idx =
+      [ WidgetNameWalletSendAddress idx
+      , WidgetNameWalletSendAmount idx
+      , WidgetNameWalletSendRemove idx
+      ]
+
+
+addOutput :: Monad m => StateT WalletWidgetState (StateT (WidgetInfo WalletWidgetState p) m) ()
+addOutput = do
+  idx <- walletSendNextOutputL <<+= 1
+  walletSendOutputsL %= Map.insert idx (WalletSendOutput "" "")
+  updateFocusList
+
+  lift $ do
+    addWidgetChild (WidgetNameWalletSendAddress idx) $
+      initEditWidget $ widgetParentLens $ walletSendOutputsL . at idx . unsafeFromJust . walletSendAddressL
+    addWidgetChild (WidgetNameWalletSendAmount idx) $
+      initEditWidget $ widgetParentLens $ walletSendOutputsL . at idx . unsafeFromJust . walletSendAmountL
+    addWidgetChild (WidgetNameWalletSendRemove idx) $
+      initButtonWidget "-"
+    addWidgetEventHandler (WidgetNameWalletSendRemove idx) $ \case
+      WidgetEventButtonPressed -> removeOutput idx
+      _ -> return ()
+
+removeOutput :: Int -> WidgetEventM WalletWidgetState p ()
+removeOutput idx = do
```

Comment:
what happens to the widgets with their handlers, etc? do we need to remove them as well?

Reply:
We should implement proper children removal at some point, but currently they don't do any harm, as they don't get any focus, so can't handle any events.